mongo/src/main/java/org/axonframework/extensions/mongo/eventhandling/deadletter/MongoSequencedDeadLetterQueue.java

```diff
@@ -0,0 +1,651 @@
+        builder.validate();
+        this.processingGroup = builder.processingGroup;
+        this.converters = builder.converters;
+        this.eventConfiguration = builder.eventConfiguration;
+        this.maxSequences = builder.maxSequences;
+        this.maxSequenceSize = builder.maxSequenceSize;
+        this.mongoTemplate = builder.mongoTemplate;
+        this.serializer = builder.serializer;
+        this.claimDuration = builder.claimDuration;
+        ensureDeadLetterIndexes(mongoTemplate.deadLetterCollection());
+    }
+
+    /**
+     * Creates a new builder, capable of building a {@link MongoSequencedDeadLetterQueue} according to the provided
+     * configuration. Note that the {@link MongoSequencedDeadLetterQueue.Builder#processingGroup(String)},
+     * {@link MongoSequencedDeadLetterQueue.Builder#mongoTemplate} and
+     * {@link MongoSequencedDeadLetterQueue.Builder#serializer(Serializer)} are mandatory for the queue to be
+     * constructed.
+     *
+     * @param <M> An implementation of {@link Message} contained in the {@link DeadLetter dead-letters} within this
+     *            queue.
+     * @return The builder
+     */
+    public static <M extends EventMessage<?>> MongoSequencedDeadLetterQueue.Builder<M> builder() {
+        return new MongoSequencedDeadLetterQueue.Builder<>();
+    }
+
+    @Override
+    public void enqueue(@Nonnull Object sequenceIdentifier, @Nonnull DeadLetter<? extends M> letter)
+            throws DeadLetterQueueOverflowException {
+        String stringSequenceIdentifier = toStringSequenceIdentifier(sequenceIdentifier);
+        if (isFull(stringSequenceIdentifier)) {
+            throw new DeadLetterQueueOverflowException(
+                    "No room left to enqueue [" + letter.message() + "] for identifier ["
+                            + stringSequenceIdentifier + "] since the queue is full."
+            );
+        }
+
+        Optional<Cause> optionalCause = letter.cause();
+        if (optionalCause.isPresent()) {
+            logger.info("Adding dead letter [{}] because [{}].", letter.message(), optionalCause.get());
+        } else {
+            logger.info("Adding dead letter [{}] because the sequence identifier [{}] is already present.",
+                        letter.message(), stringSequenceIdentifier);
+        }
+
+        DeadLetterEventEntry entry = converters
+                .stream()
+                .filter(c -> c.canConvert(letter.message()))
+                .findFirst()
+                .map(c -> c.convert(letter.message(), serializer))
+                .orElseThrow(() -> new NoMongoConverterFoundException(
+                        String.format("No converter found for message of type: [%s]",
+                                      letter.message().getClass().getName()))
+                );
+
+        EventEntryConfiguration configuration = EventEntryConfiguration.getDefault();
+        Document message = entry.asDocument(configuration);
```

Comment:
```suggestion
        Document message = entry.asDocument(eventConfiguration);
```
You are not using the instance variable, but always the default. Any customization is lost. 
Remove the line above and change the variable passed into this method